scripts/ci/check_compliance.py

```diff
@@ -491,6 +491,77 @@ def required_false_check(self, binding):
                     "'required: false' is redundant, please remove"
                 )
 
+
+class DevicetreeLintingCheck(ComplianceTest):
+    """
+    Checks if we are introducing syntax or formatting issues to devicetree files.
+    """
+    name = "DevicetreeLinting"
+    doc = "See https://docs.zephyrproject.org/latest/contribute/style/devicetree.html for more details."
+
+    def _parse_json_output(self, cmd, cwd=None):
+        """Run command and parse single JSON output with issues array"""
+        result = subprocess.run(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+            text=True,
+            cwd=cwd or GIT_TOP
+        )
+
+        if result.stdout.strip():
+            json_data = json.loads(result.stdout)
+            return json_data
+        else:
+            return None
+
+    def run(self):
+        # Get changed DTS files
+        dts_files = [
+            file for file in get_files(filter="d")
+            if file.endswith((".dts", ".dtsi", ".overlay"))
+        ]
+
+        if not dts_files:
+            self.skip('No DTS')
+
+        cmd = ["npx", "dts-linter", "--outputFormat", "json", "--format", "--patchFile", "dts_linter.patch"]
```

Comment:
Do not use `npx` here, it will download `dts-linter` if not found, which is exactly what we don't want to prevent supply chain attacks.

Not sure how you can ensure `dts-linter` is in `$PATH`, you could add a script to `package.json`:

```json
{
  "private": true,
  "dependencies": {
    "dts-linter": "^0.3.0-beta2"
  },
  "scripts": {
    "dts-linter": "dts-linter"
  }
}
```

And run
```shell
$ npm run dts-linter
```